src/main/java/com/example/shoppingonlybackend/controller/ProductController.java

```diff
@@ -68,4 +68,25 @@ public ResponseEntity<?> delete(@PathVariable int id) {
         }
         return ResponseEntity.notFound().build();
     }
+    
+    @PreAuthorize("hasRole('MEMBER') or hasRole('ADMIN')")
+    @GetMapping("/searchByName")
+    public ResponseEntity<?> searchByName(@RequestParam String keyword) {
+        List<Product> products = productService.searchProductsByName(keyword);
```

Comment:
```suggestion

        List<Product> products = productService.searchProductsByName(keyword);
```
Metoda girişte 1 satır boş bırakalım.